Kitodo-DataAccess/src/test/java/org/kitodo/dataaccess/storage/memory/MemoryNodeTest.java

```diff
@@ -0,0 +1,947 @@
+    }
+
+    /**
+     * Tests {@code isEmpty()} for an empty node.
+     */
+    @Test
+    public void testIsEmpty() {
+        assertThat(new MemoryNode().isEmpty(), is(true));
+    }
+
+    /**
+     * Tests {@code isEmpty()} for a non-empty node.
+     */
+    @Test
+    public void testIsNotEmpty() {
+        assertThat(new MemoryNode(Mets.METS).isEmpty(), is(false));
+    }
+
+    /**
+     * Tests {@code iterator()}.
+     */
+    @Test
+    public void testIterator() {
+        MemoryNode node = new MemoryNode(Mods.NAME);
+        node.add(new MemoryNode("http://names.example/alice"));
+        node.add(new MemoryNamedNode("http://names.example/bob"));
+        node.add(new MemoryNodeReference("http://names.example/charlie"));
+
+        Set<ObjectType> expected = new HashSet<>();
+        expected.add(new MemoryNode("http://names.example/alice"));
+        expected.add(new MemoryNamedNode("http://names.example/bob"));
+        expected.add(new MemoryNodeReference("http://names.example/charlie"));
+
+        Iterator<ObjectType> i = node.iterator();
+        while (i.hasNext()) {
+            expected.remove(i.next());
+        }
+        assertThat(expected.isEmpty(), is(true));
+    }
+
+    /**
+     * Tests {@code last()}.
+     */
+    @Test
+    public void testLast() {
+        MemoryNode node = new MemoryNode(Mods.NAME);
+        node.add(new MemoryNode("http://names.example/alice"));
+        node.add(new MemoryNode("http://names.example/bob"));
+        node.add(new MemoryNode("http://names.example/charlie"));
+
+        assertEquals((long) 3, (long) node.last().orElse(null));
+    }
+
+    /**
+     * Tests {@code matches(ObjectType)} for a matching candidate.
+     */
+    @Test
+    public void testMatches() {
+        MemoryNode node = new MemoryNode(Mods.NAME);
+        node.add(new MemoryNode("http://names.example/alice"));
+        node.add(new MemoryNode("http://names.example/bob"));
+        node.add(new MemoryNode("http://names.example/charlie"));
+
+        MemoryNode condition = new MemoryNode().put(RDF.toURL(2), new MemoryNode("http://names.example/bob"));
+
+        assertThat(node.matches(condition), is(true));
+    }
+
+    /**
+     * Tests {@code matches(ObjectType)} for a non-matching candidate.
+     */
+    @Test
+    public void testMatchesNot() {
+        MemoryNode node = new MemoryNode(Mods.NAME);
+        node.add(new MemoryNode("http://names.example/alice"));
+        node.add(new MemoryNode("http://names.example/bob"));
+        node.add(new MemoryNode("http://names.example/charlie"));
+
+        MemoryNode condition = new MemoryNode(Mods.NAME).put(RDF.toURL(9),
+            new MemoryNode("http://names.example/alice"));
+
+        assertThat(node.matches(condition), is(false));
+    }
+
+    /**
+     * Tests {@code MemoryNode()} constructor.
+     */
+    @Test
+    public void testMemoryNodeCanBeCreatedEmpty() {
+        new MemoryNode();
```

Comment:
What are you testing here?

Reply:
The constructor. It initializes an object and does not throw an exception.